### ISSUE TITLE
[Bug]: Synchronization does not run on Webdav server with subpath

### ISSUE THREAD

**Issue by cyruz-git** [remotely-save/remotely-save#825]
### What happened?

Running the synchronization on a Webdav server using a subpath (eg: https://webdav.example.com/somepath/) produces the following error:

`Error: "/../Notes" doesn't starts with "/Notes/"`

I see that [PR 226](https://github.com/remotely-save/remotely-save/pull/226) was meant to fix this, but has been reverted. Any chance this can be fixed?

I'm using Caddy v2 as webdav server, please let me know if you need to reproduce the error.

### What OS are you using?

macOS

### What remote cloud services are you using? (Please choose the specified one if it's in the list)

webdav

### Version of the plugin

0.5.21

### Version of Obsidian

1.6.7

### Using password or not

- [ ] Yes.

### Ensure no sensitive information

- [X] I ensure that no sensitive information is submitted in the issue.

**Comment by fyears** [comment]
Yeah the point is I don't know how to reproduce the error.

Could you please show me how to reproduce the error? Especially how to setup your server and what's the input in you remotely save settings? (address, remote base path)

**Comment by cyruz-git** [comment]
Sure. I suggest you try downloading a [Caddy binary](https://caddyserver.com/download) adding the webdav module [http.handlers.webdav](https://caddyserver.com/docs/modules/http.handlers.webdav). Just search for `webdav` in the download page, select the module and start the download.

Then run Caddy with a Caddyfile similar to this (adjust the root to the folder you want to serve):

``` 
http://localhost:7070 {
	route /dav/* {
		basicauth {
			user $2a$14$3v.HuXc0R/DcIk8eRnjWfeuVLrMMcbA5DmIfOKzCUyxs0.SnrGyQS
		}
		root /Users/youruser/webdav
		uri strip_prefix /dav
		webdav
	}
}
``` 

Place the Caddyfile in the same folder of the Caddy binary, so you can run everything with a simple:

`Caddy run`

In RemotelySave, configure the plugin with the following data:

> Remote Service: Webdav
> Server Address: http://localhost:7070/dav
> Username: user
> Password: password
> Auth Type: basic

Then to trigger the error, just run the sync.

**Comment by fyears** [comment]
Hi!

1. I found the bug and fix it in 0.5.25. Please try.
2. And I also modify your instruction about caddy and post it here with credit: https://github.com/remotely-save/remotely-save/blob/master/docs/remote_services/webdav_caddy/README.md  so that other users can know how to configure it. Hope you don't mind it. Thank you for your contributions!

**Comment by cyruz-git** [comment]
Hi @fyears, thanks, it works fine now.

Thanks for the mention, I'm happy to be able to contribute in some ways.
